src/data/mod.rs

```diff
@@ -622,3 +622,172 @@ pub fn replace_scalar(a: &mut Array, cond: &Array, b: f64) {
         HANDLE_ERROR(AfError::from(err_val));
     }
 }
+
+/// Create an Array of particular shape filled with given constant.
+///
+/// You can use this function to create arrays of the following types only.
+/// - DType::F32
+/// - DType::C32
+/// - DType::F64
+/// - DType::C64
+/// - DType::B8
+/// - DType::S32
+/// - DType::U32
+/// - DType::U8
+/// - DType::S16
+/// - DType::U16
+///
+/// # Parameters
+///
+/// - `real` is the constant with which output Array is to be filled
+/// - `imag` is the constant with which imaginary part of a complex Array will be filed.
+/// - `dims` is the size of Array
+/// - `dtype` indicates whats the type of the Array to be created
+///
+/// # Return Values
+///
+/// Array of `dims` shape and filed with given constant `value`.
+pub fn create_constant(real: f64, imag: f64, dims: Dim4, dtype: DType) -> Array {
```

Comment:
Is it possible to remove all the `create_` parts from these functions? I think it is obvious that someone wants to create a constant if they call the `constant` function. It will also be more more consistent with the [AF api](http://arrayfire.org/docs/group__data__func__constant.htm)